app/views/teachers/TeacherStudentView.coffee

```diff
@@ -0,0 +1,392 @@
+
+    @levelData = []
+    for versionedCourse in @classroom.get('courses') ? []
+      course = _.find @courses.models, (c) => c.id is versionedCourse._id
+      for versionedLevel in versionedCourse.levels
+        @playTime = 0 # this and @timesPlayed should probably only count when the levels are completed
+        @timesPlayed = 0
+        @studentTime = 0
+        @levelProgress = 'not started'
+        for session in @sessions.models
+          if session.get('level').original == versionedLevel.original
+            # if @levelProgressMap[versionedLevel.original] == 'complete' # ideally, don't log sessions that aren't completed in the class
+            @playTime += session.get('playtime') or 0
+            @timesPlayed += 1
+            if session.get('creator') is @studentID
+              @studentTime = session.get('playtime') # this can be null, apparently.
+              if @levelProgressMap[versionedLevel.original] == 'complete'
+                @levelProgress = 'complete'
+              else if @levelProgressMap[versionedLevel.original] == 'started'
+                @levelProgress = 'started'
+        classAvg = if @timesPlayed and @timesPlayed > 0 then Math.round(@playTime / @timesPlayed) else 0 # only when someone other than the user has played
+        console.log (@timesPlayed)
+        @levelData.push {
+          levelID: versionedLevel.original
+          levelIndex: @classroom.getLevelNumber(versionedLevel.original)
+          levelName: versionedLevel.name
+          courseName: course.get('name')
+          courseID: course.get('_id')
+          class: classAvg
+          student: if @studentTime then @studentTime else 0
+          levelProgress: @levelProgress
+          # required:
+        }
+    # console.log (@levelData)
+
+    # new map for averages of all classroom playtimes
+    # @averageLevelPlaytimeMap = {}
+    #   for versionedCourse in @classroom.get('courses') ? []
+    #     for versionedLevel in versionedCourse.levels
+    #       var totalPlaytime = null
+    #       var totalSessions = 0
+    #       session = @levelSessionMap[versionedLevel.original]
+    #       # get playtimes for each session in all classroom sessions of this course with this particular level
+    #       if totalPlaytime
+    #         var averagePlaytime = totalPlaytime / totalSessions
+    #         @averagePlaytimeMap[versionedLevel.original] = averagePlaytime
+
+  studentStatusString: () ->
+    status = @user.prepaidStatus()
+    expires = @user.get('coursePrepaid')?.endDate
+    string = switch status
+      when 'not-enrolled' then $.i18n.t('teacher.status_not_enrolled')
+      when 'enrolled' then (if expires then $.i18n.t('teacher.status_enrolled') else '-')
+      when 'expired' then $.i18n.t('teacher.status_expired')
+    return string.replace('{{date}}', moment(expires).utc().format('l'))
+
+  onClickEnrollStudentButton: (e) ->
+    userID = $(e.currentTarget).data('user-id')
```

Comment:
For reference: `data-user-id` is a good way to attach ID numbers to elements, and this is how you get that info back out. It might be better to use this than inserting it into the `id` attribute.